File: src/events/guildMemberRemove.js
const { Events } = require('discord.js');
const settings = require('../../config/settings.json');

module.exports = {
    name: Events.GuildMemberRemove,
    async execute(member) {
        // Solo actualizar si es el servidor configurado
        if (member.guild.id !== settings.guildId) return;

        // Solo actualizar si el miembro que salió no era un bot
        if (member.user.bot) return;

        member.client.updatePresenceCount({ delta: -1 }).catch(() => {});
    },
};

File: pybot/events/guild_member_remove.py
from __future__ import annotations

from discord.ext import commands

from .. import config


class GuildMemberRemoveEvents(commands.Cog):
    def __init__(self, bot: commands.Bot) -> None:
        self.bot = bot

    @commands.Cog.listener()
    async def on_member_remove(self, member):
        settings = config.resolve_config()
        if not settings.get("guildId"):
            return
        if str(member.guild.id) != str(settings.get("guildId")):
            return
        if member.bot:
            return
        await self.bot.update_presence_count(delta=-1)


async def setup(bot: commands.Bot) -> None:
    await bot.add_cog(GuildMemberRemoveEvents(bot))
